core/src/main/java/com/linecorp/armeria/server/RouteResult.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.armeria.common.MediaType;
+
+/**
+ * The result returned by {@link Route#apply(RoutingContext)}.
+ */
+public class RouteResult {
+
+    static final int LOWEST_SCORE = Integer.MIN_VALUE;
+    static final int HIGHEST_SCORE = Integer.MAX_VALUE;
+
+    private static final RouteResult EMPTY =
+            new RouteResult(PathMappingResult.empty(), LOWEST_SCORE, null);
+
+    /**
+     * The {@link RouteResult} whose {@link #isPresent()} returns {@code false}. It is returned by
+     * {@link Route#apply(RoutingContext)} when the {@link RoutingContext} did not match the
+     * conditions in the {@link Route}.
+     */
+    public static RouteResult empty() {
+        return EMPTY;
+    }
+
+    /**
+     * Returns a new builder.
+     */
+    public static RouteResultBuilder builder() {
+        return new RouteResultBuilder();
+    }
+
+    private final PathMappingResult pathMappingResult;
```

Comment:
I'm curious if we can merge `PathMappingResult(Builder)` into `RoutingResult(Builder)` somehow, because there's always 1-to-1 cardinality for each other. For example, `PathMapping.apply()` could return a `RoutingResultBuilder`? Not sure if it is a very good idea though. Thoughts?

Reply:
This is not solved yet, right?

Reply:
Didn't have time to solve it yet :)

Reply:
Yeah, it seems fine that `PathMapping.apply()` returns a `RoutingResultBuilder`. Let me remove the `PathMappingResult`. :-)